### PR TITLE
[Clang][Sema] Add a test for move ctor calling for a base class. NFC

### PR DESCRIPTION
**Pull request by samolisov**
When clang compiles the following expression:

```c++
  return A{B{"Move Ctor"}};
```
(where `B` is a base class for `A`), it adds a call to the move constructor of `B`. When the code is changed to...

```c++
  return A{{"No Move Ctor"}};
```
... a move constructor is invoked neither for `A` nor for `B`.

The lit test demonstrates the difference in the generated AST.

### CODE REVIEW

**Comment by samolisov** [comment]
This is a pre-fix test (but, actually, I have no fix for now) to demonstrate the difference in compilation between some looking very similar pieces of the code. I sure the root cause (if this is an issue at all) lies in `SemaInit.cpp`, this is why I've added the developers who were working with this file recently as reviewers. Because this is my first lit test for clang, I'm not sure `AST` is the correct directory for such kind of tests but I didn't manage to find a better place: all the tests in the `SemaCXX` as well as in the `Parser` directories check some error messages or work with some IR, not AST dumps. If you think there is more appropriate directory for the test, please let me know.

File: clang/test/AST/explicit-base-class-move-cntr.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// RUN: %clang_cc1 -ast-dump -fblocks %s | FileCheck -strict-whitespace %s
```
Comment (vitalybuka) [inline clang/test/AST/explicit-base-class-move-cntr.cpp:1]: can this test be autogenerated?

Comment (samolisov) [reply clang/test/AST/explicit-base-class-move-cntr.cpp:1]: Unfortunately, I have no idea how to generate tests for clang. This test has been manually written.

Comment (vitalybuka) [reply clang/test/AST/explicit-base-class-move-cntr.cpp:1]: I was thinking about utils/update_cc_test_checks.py
But this is AST test

Maybe llvm-project/clang/test/AST/gen_ast_dump_json_test.py

Comment (samolisov) [reply clang/test/AST/explicit-base-class-move-cntr.cpp]: @vitalybuka Thank you for the suggestion. I've regenerated the test using the tool and cut the dump to make it a bit shorter.

**Comment by samolisov** [comment]
Gentle ping.

I've addressed all the comments (thank you, @vitalybuka).

**Comment by vitalybuka** [comment]
> Gentle ping.

Oh, you should click "re-request review", so it's popup in github review query.

Looking.

**Review by vitalybuka — APPROVED** [review]
LGTM, but probably someone who contribute these components more should take a look.

**Pull request merged**